webapp/src/components/modals/create_issue/create_issue.jsx

```diff
@@ -0,0 +1,183 @@
+
+import FormButton from 'components/form_button';
+import GithubRepoSelector from 'components/github_repo_selector';
+import Input from 'components/input';
+
+const initialState = {
+    submitting: false,
+    error: null,
+    repoValue: '',
+    issueTitle: '',
+};
+
+export default class CreateIssueModal extends PureComponent {
+    static propTypes = {
+        close: PropTypes.func.isRequired,
+        create: PropTypes.func.isRequired,
+        post: PropTypes.object,
+        theme: PropTypes.object.isRequired,
+        visible: PropTypes.bool.isRequired,
+    };
+
+    constructor(props) {
+        super(props);
+        this.state = initialState;
+    }
+
+    // handle issue creation after form is populated
+    handleCreate = (e) => {
+        if (e && e.preventDefault) {
+            e.preventDefault();
+        }
+
+        if (!this.state.issueTitle) {
+            return;
+        }
+
+        // const title = this.state.issueTitle;
+        // const description = this.props.post.message;
+        // const repo = this.state.repoValue;
+        // const repoUrlParts = this.state.repoValue.split('/');
+        // const repo = repoUrlParts.pop();
+
+        const issue = {
+            title: this.state.issueTitle,
+            body: this.props.post.message,
+            repo: this.state.repoValue.name,
+        };
+
+        this.setState({submitting: true});
+
+        this.props.create(issue).then((created) => {
+            if (created.error) {
+                this.setState({error: created.error.response.body.message, submitting: false});
+                return;
+            }
+            this.handleClose(e);
+        });
+    };
+
+    handleClose = (e) => {
+        if (e && e.preventDefault) {
+            e.preventDefault();
+        }
+        const {close} = this.props;
+        this.setState(initialState, close);
+    };
+
+    handleRepoValueChange = (newValue) => {
+        this.setState({
+            repoValue: newValue,
+        });
+    };
+
+    handleIssueTitleChange = (newValue) => {
+        this.setState({
+            issueTitle: newValue,
+        });
+    };
+
+    render() {
+        const {visible, theme} = this.props;
+        const {error, submitting} = this.state;
+        const style = getStyle(theme);
+
+        if (!visible) {
+            return null;
+        }
+
+        const component = (
+            <div>
+                <GithubRepoSelector
+                    id={'repo'}
+                    onChange={this.handleRepoValueChange}
+                    required={true}
+                    theme={theme}
+                    error={error}
+                    value={this.state.repoValue}
+                />
+                <Input
+                    id={'title'}
+                    label='Title for the GitHub Issue'
+                    type='input'
+                    required={true}
+                    disabled={false}
+                    readOnly={false}
```

Comment:
We can omit these as they are default values.

Reply:
@maria, I agree that `readOnly` defaults to `false`, but I don't see `disabled` defaulting to any value inside https://github.com/dhadiseputro/mattermost-plugin-github/blob/b53920708ee1e927f270e1110745ef98b24fbdb6/webapp/src/components/input.jsx

Reply:
The `disabled` prop is set internally to the `input` tag where it is a boolean and by default is false. 

However, this is a non-blocking nit.

Reply:
Ah, yes. I was checking only the react `Input` component and didn't propagate into the actual defaults set by the `input` tag.

Reply:
ticket created. https://github.com/mattermost/mattermost-plugin-github/issues/205